File: library/src/main/java/com/github/alexjlockwood/kyrie/ClipType.java
package com.github.alexjlockwood.kyrie;

import android.support.annotation.IntDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/** Determines the clipping strategy of a {@link ClipPathNode}. */
@Retention(RetentionPolicy.SOURCE)
@IntDef({ClipType.INTERSECT, ClipType.SUBTRACT})
public @interface ClipType {
  /** Only the pixels drawn inside the bounds of the clip path will be displayed. */
  int INTERSECT = 0;
  /** Only the pixels drawn outside the bounds of the clip path will be displayed. */
  int SUBTRACT = 1;
}

File: kyrie/src/main/java/com/github/alexjlockwood/kyrie/ClipType.kt
package com.github.alexjlockwood.kyrie

/** Determines the clipping strategy of a [ClipPathNode]. */
enum class ClipType {
    /** Only the pixels drawn inside the bounds of the clip path will be displayed.  */
    INTERSECT,
    /** Only the pixels drawn outside the bounds of the clip path will be displayed.  */
    SUBTRACT
}
